projects/gateway2/translator/gateway_translator.go

```diff
@@ -82,17 +90,28 @@ func (t *translator) TranslateProxy(
 		reporter.Gateway(gateway).Listener(&listener).SetAttachedRoutes(uint(availRoutes))
 	}
 
+	for _, ls := range consolidatedGateway.AllowedListenerSets {
```

Comment:
could this be some sort of util in `query` that takes a reporter, consolidatedGateway and a routesForGw result?

Reply:
Updated to its own method